### PR TITLE
Allowed CCX coaches to see individual students progress from grade book

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
### Background
https://github.com/mitocw/edx-platform/issues/88

### What is done in this PR
**Studio Updates:** None.
**LMS Updates:** 
 - Coach can open grade book and click on student name to see his progress on CCX.
 
@pdpinch @giocalitri 

- Enrollment tab on coach dashboard.
screen shot 2015-12-29 at 5 32 23 pm

- Coach can open grade book from Student Admin tab from his dashboard
screen shot 2015-12-29 at 5 32 33 pm

- On grade book coach can click on student name to see his progress.
screen shot 2015-12-29 at 5 32 40 pm

### CODE REVIEW

File: lms/djangoapps/courseware/access.py [hunk 1]
```diff
@@ -62,9 +63,37 @@
...
+    Check if user is a coach on this ccx.
+
+    Arguments:
+        user (User): the user whose descriptor access we are checking.
+        course_key (CCXLocator): Key to CCX.
+
+    Returns:
+        bool: whether user is a coach on this ccx or not.
+    """
+    if isinstance(course_key, CCXLocator):
```
Comment (pdpinch) [inline lms/djangoapps/courseware/access.py]: Would it make more sense to raise an exception if course_key isn't a CCX locator?

File: lms/djangoapps/courseware/access.py [hunk 2]
```diff
@@ -62,9 +63,37 @@
...
+        role = CourseCcxCoachRole(course_key)
+
+        if role.has_user(user):
+            list_ccx = CustomCourseForEdX.objects.filter(
+                course_id=course_key.to_course_locator(),
+                coach=user
+            )
+            if list_ccx.exists():
+                coach_ccx = list_ccx[0]
+                return long(coach_ccx.id) == long(ccx_id)
```
Comment (pdpinch) [inline lms/djangoapps/courseware/access.py]: Is this type conversation necessary?

File: lms/djangoapps/courseware/tests/test_access.py [hunk 3]
```diff
@@ -44,9 +56,96 @@
...
 )
 
+from lms.djangoapps.ccx.models import CustomCourseForEdX
+
 # pylint: disable=protected-access
 
 
+class CoachAccessTestCaseCCX(SharedModuleStoreTestCase, LoginEnrollmentTestCase):
+    """
+    Test is user is coach on ccx.
```
Comment (pdpinch) [inline lms/djangoapps/courseware/tests/test_access.py]: typo `Test if user is coach on ccx.`

File: lms/djangoapps/courseware/views.py [hunk 4]
```diff
@@ -920,13 +920,15 @@ def _progress(request, course_key, student_id):
         return redirect(reverse('course_survey', args=[unicode(course.id)]))
 
     staff_access = bool(has_access(request.user, 'staff', course))
+    coach_access = has_ccx_coach_role(request.user, course_key)
+    has_access_on_students_profiles = staff_access or coach_access
```
Comment (pdpinch) [inline lms/djangoapps/courseware/views.py]: We'll have to run this by Dave Ormsbee to get his input on it.

File: lms/djangoapps/courseware/tests/test_access.py [hunk 5]
```diff
@@ -44,9 +56,96 @@
     seed_milestone_relationship_types,
 )
 
+from lms.djangoapps.ccx.models import CustomCourseForEdX
+
 # pylint: disable=protected-access
 
 
+class CoachAccessTestCaseCCX(SharedModuleStoreTestCase, LoginEnrollmentTestCase):
```
Comment (pdpinch) [inline lms/djangoapps/courseware/tests/test_access.py]: How is it that we didn't have this test before?

Comment (amir-qayyum-khan) [inline lms/djangoapps/courseware/tests/test_access.py]: @pdpinch 
I do not know. But I am sure we do not have access tests for ccx in coursware.
2nd thing is that I just incorporate coach role to courseware access to allow him to see progress of individual like staff.

**Should I create this PR on upstream for Ormsbee input?

File: lms/djangoapps/courseware/access.py (hunk 2 above)
Comment (amir-qayyum-khan) [inline lms/djangoapps/courseware/access.py]: because ```ccx_id``` is a string and ```coach_ccx.id``` is number

**Comment by amir-qayyum-khan** [comment]
@pdpinch done with ur feedback. Should I create this PR to upstream?

**Comment by pdpinch** [comment]
Thanks. This is functioning well for me. I'm not sure how edX will respond though. Let's open it upstream and find out. 

We should probably get Justin or Giovanni to look at it as well.

**Comment by amir-qayyum-khan** [comment]
Made upstream PR https://github.com/edx/edx-platform/pull/11124

**Pull request closed without merging**